### PR TITLE
Flesh out the info cog/command to be more expansible

### PR DESCRIPTION
**Pull request by AbstractUmbra**
The groundwork done in #50 was solid but it felt very restrictive. I have stretched this out into factory methods and more so it can be expanded upon easily.

### CODE REVIEW

File: modules/info.py [hunk 1]
```diff
@@ -22,57 +22,117 @@
...
+
+    async def cog_command_error(self, ctx: core.Context, error: commands.CommandError) -> None:  # type: ignore # bad lib types.
+        error = getattr(error, "original", error)
+
+        if isinstance(error, NoPermissions):
+            await ctx.send("Sorry, you don't have permissions to view details on this object.")
+            return
+
+    def _embed_factory(self, entity: EntityT) -> discord.Embed:
+        embed = discord.Embed(title=f"Info on {entity.name}!", colour=discord.Colour.random())
```
Comment (EvieePy) [inline modules/info.py:55]: Why are we using a random colour?

File: modules/info.py [hunk 2]
```diff
@@ -22,57 +22,117 @@
...
+        embed.add_field(name="ID:", value=entity.id)
+        embed.timestamp = discord.utils.utcnow()
```
Comment (EvieePy) [inline modules/info.py:57]: Shouldn't we be using `.now(timezone.utc)`

File: modules/info.py [hunk 3]
```diff
@@ -22,57 +22,117 @@
...
+    def _user_info(self, user: discord.User, /, *, embed: discord.Embed) -> discord.Embed:
+        embed = discord.Embed(title=f"Info on {user.display_name}!", colour=discord.Colour.random())
+        embed.set_author(name=user.name)
+        embed.set_image(url=user.display_avatar.url)
+        created_at_fmt = (
+            discord.utils.format_dt(user.created_at, "F") + "\n" f"({discord.utils.format_dt(user.created_at, 'R')})"
+        )
+        embed.add_field(name="Account was created on:", value=created_at_fmt)
+
+        embed.timestamp = discord.utils.utcnow()
```
Comment (EvieePy) [inline modules/info.py:94]: Shouldn't we be using `.now(timezone.utc)`

**Review by EvieePy — CHANGES_REQUESTED** [review]

File: modules/info.py (hunk 2 above)
Comment (EvieePy) [reply modules/info.py:57]: Nevermind chilly pointed out this was dpy not datetime 😆

File: modules/info.py (hunk 1 above)
Comment (AbstractUmbra) [reply modules/info.py:55]: It uses a random colour by default but certain factory methods do change it.
You can see that in `_member_info` and `_role_info` we take their colour if present.

If you'd prefer a more static default colour I'm happy to oblige.

**Review by EvieePy — APPROVED** [review]

**Pull request merged**